server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -264,6 +271,44 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
+    private Map<String, DocumentField> readFields(StreamInput in) throws IOException {
```

Comment:
Reading and writing the document fields directly shoudln't be necessary. Instead we can use `in.readMap(StreamInput::readString, DocumentField::new)` and for writing `out.writeMap(map, StreamOutput::writeString, DocumentField::writeTo)`

Reply:
added readMap/writeMap for communication with server versions 8.0.0+; for communication with the older versions the old serialization using explicit DocumentField parsing remains as before to avoid potential backward compatibility bugs.